Add notify entity parameter for Telegram bot actions (#43382)
home-assistant/home-assistant.github.io: source/_integrations/telegram_bot.markdown

Available actions: `send_message`, `send_photo`, `send_video`, `send_animation`, `send_voice`, `send_sticker`, `send_document`, `send_location`, `send_chat_action`, `edit_message`, `edit_message_media`, `edit_caption`, `edit_replymarkup`, `answer_callback_query`, `delete_message`, `leave_chat` and `set_message_reaction`.

> Targets can be specified in any of the following ways:

Reviewer: I think we should avoid talking about "target" on its own. It's confusing what we mean with that, considering the old parameter and that it's a common term elsewhere in Home Assistant actions.
```suggestion
Chat targets can be specified in any of the following ways:
```

- `entity_id`
- `config_entry_id` and `chat_id`

> - If you only have 1 bot and no targets were specified, the bot's first subentry will be used as the default target.

Reviewer: ```suggestion
- If you only have 1 bot and no chat targets (`entity_id` or `chat_id`) were specified, the bot's first subentry will be used as the default chat.
```

| Data attribute     | Optional | Description                                                                                                                                                                                                                                                                                               |
| -------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the message. |

Reviewer: ```suggestion
| `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the message. |
```

| Data attribute | Optional | Description                                                                                                                                                                                                                                                                                               |
| ---------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the photo. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the photo. |
```

| Data attribute | Optional | Description                                                                                                                                                                                                                                                                                               |
| ---------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the video. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the video. |
```

| Data attribute | Optional | Description                                                                                                                                                                                                                                                                                               |
| ---------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the animation. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the animation. |
```

| Data attribute | Optional | Description                                                                                                                                                                                                                                                                                               |
| ---------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the voice message. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the voice message. |
```

| Data attribute | Optional | Description                                                                                                                                                                                                                                                                                               |
| ---------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the sticker. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the sticker. |
```

| Data attribute | Optional | Description                                                                                                                                                                                                                                                                                               |
| ---------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the document. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the document. |
```

| Data attribute | Optional | Description                                                                                                                                                                                                                                                                                               |
| ---------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the location. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the location. |
```

| Data attribute    | Optional | Description                                                                                                                                                                    |
| ------------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | `entity_id`               | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the poll. |

Reviewer: ```suggestion
| `entity_id`               | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the poll. |
```

| Data attribute     | Optional | Description                                                                                                                                                                                                                                                                                               |
| -------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the chat action. |

Reviewer: ```suggestion
| `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the chat action. |
```

| Data attribute     | Optional | Description                                                                                                                                                                                                                                                                                               |
| -------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and target for editing the message. |

Reviewer: ```suggestion
| `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for editing the message. |
```

| Data attribute     | Optional | Description                                                                                                                                                                                                                                                                                               |
| -------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for editing the message media. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for editing the message media. |
```

| Data attribute     | Optional | Description                                                                                                                                                                                                                                                                                               |
| -------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and target for editing the caption. |

Reviewer: ```suggestion
| `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for editing the caption. |
```

| Data attribute     | Optional | Description                                                                                                                                                                                                                                                                                               |
| -------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and target for editing the inline keyboard. |

Reviewer: ```suggestion
| `entity_id`                | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for editing the inline keyboard. |
```

| Data attribute | Optional | Description                                                                                                                                                                                                                                                |
| ---------------------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for deleting the message. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for deleting the message. |
```

| `config_entry_id`      | yes      | The config entry representing the Telegram bot to delete the message. Required if you have multiple Telegram bots.|
| `chat_id`              | yes      | The chat_id where to delete the message.                                                                                                                                                                                                                   |
| `message_id`           | no       | Id of the message to delete. When answering a callback from a pressed button, the id of the origin message is in: {% raw %}`{{ trigger.event.data.message.message_id }}`{% endraw %}. You can use `"last"` to refer to the last message sent to `chat_id`. |

### Action `telegram_bot.leave_chat`

Remove the bot from the chat group where it was added.

| Data attribute         | Optional | Description                               |
| ---------------------- | -------- | ----------------------------------------- |

> | `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and target for leaving the chat. |

Reviewer: ```suggestion
| `entity_id`            | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for leaving the chat. |
```

| `config_entry_id`      | yes      | The config entry representing the Telegram bot to leave the chat. Required if you have multiple Telegram bots.|
| `chat_id`              | yes      | The chat_id from where to remove the bot. |

### Action `telegram_bot.set_message_reaction`

Sets the bot's reaction for a given message.

| Data data attribute | Optional | Description                                                      |
| ------------------- | -------- | ---------------------------------------------------------------- |

> | `entity_id`         | yes      | Notify entities where each entity has its corresponding Telegram bot and target for sending the message. |

Reviewer: ```suggestion
| `entity_id`         | yes      | Notify entities where each entity has its corresponding Telegram bot and chat for sending the message. |
```
